app/lib/package/name_tracker.dart

```diff
@@ -43,18 +44,26 @@ class NameTracker {
     _reducedNames.add(reducePackageName(name));
   }
 
+  void addReservedName(String name) {
+    _reservedNames.add(name);
+    _reducedNames.add(reducePackageName(name));
+  }
+
   /// Whether the package was already added to the tracker.
   bool _hasPackage(String name) => _names.contains(name);
 
-  /// Whether the [name] has a conflicting package that already exists.
+  /// Whether the [name] has a conflicting package that already exists or
+  /// a conflicting package among the moderated packages.
   bool _hasConflict(String name) =>
-      _reducedNames.contains(reducePackageName(name));
+      _reducedNames.contains(reducePackageName(name)) ||
+      _reservedNames.contains(name);
 
   /// Whether to accept the upload attempt of a given package [name].
   ///
   /// Either the package [name] should exists, or it should be different enough
-  /// from already existing package names. An example for the rejection:
-  /// `long_name` will be rejected, if package `longname` or `lon_gname` exists.
+  /// from already existing active or moderated package names. An example for
+  /// the rejection: `long_name` will be rejected, if package `longname` or
+  /// `lon_gname` exists.
```

Comment:
nit: `lon_gname` -> `long_name`

Reply:
Isn't it supposed to be `lon_gname`, or am I misunderstanding the comment?

Reply:
My bad, automatic spell check error on my part...